#### Data Source

Wine Quality Dataset

Source:
UCI Machine Learning Repository

Reference:
Cortez et al. (2009)

#### Load Data

The first step is to import the Pandas library and load the dataset into a DataFrame using the `read_csv()` method.

The Wine Quality dataset uses a semicolon `";"` as the delimiter instead of the default comma `","`. The `sep` parameter is set to `";"` to ensure that the data is separated into the correct columns and displayed properly.


In [1]:
import pandas as pd

df = pd.read_csv("../data/raw/winequality-red.csv", sep=";")

df.head()

,fixed acidity,volatile acidity,citric acid,residual sugar,chlorides,free sulfur dioxide,total sulfur dioxide,density,pH,sulphates,alcohol,quality
0,7.4,0.70,0.00,1.9,0.076,11.0,34.0,0.9978,3.51,0.56,9.4,5
1,7.8,0.88,0.00,2.6,0.098,25.0,67.0,0.9968,3.20,0.68,9.8,5
2,7.8,0.76,0.04,2.3,0.092,15.0,54.0,0.9970,3.26,0.65,9.8,5
3,11.2,0.28,0.56,1.9,0.075,17.0,60.0,0.9980,3.16,0.58,9.8,6
4,7.4,0.70,0.00,1.9,0.076,11.0,34.0,0.9978,3.51,0.56,9.4,5


In [2]:
display(df.shape)
display(df.columns)
df.info()

(1599, 12)

Index(['fixed acidity', 'volatile acidity', 'citric acid', 'residual sugar',
       'chlorides', 'free sulfur dioxide', 'total sulfur dioxide', 'density',
       'pH', 'sulphates', 'alcohol', 'quality'],
      dtype='object')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1599 entries, 0 to 1598
Data columns (total 12 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   fixed acidity         1599 non-null   float64
 1   volatile acidity      1599 non-null   float64
 2   citric acid           1599 non-null   float64
 3   residual sugar        1599 non-null   float64
 4   chlorides             1599 non-null   float64
 5   free sulfur dioxide   1599 non-null   float64
 6   total sulfur dioxide  1599 non-null   float64
 7   density               1599 non-null   float64
 8   pH                    1599 non-null   float64
 9   sulphates             1599 non-null   float64
 10  alcohol               1599 non-null   float64
 11  quality               1599 non-null   int64  
dtypes: float64(11), int64(1)
memory usage: 150.0 KB


## Initial Dataset Inspection

The red wine dataset contains **1,599 observations** and **12 variables**.

Inspection of the dataset using `df.shape`, `df.columns`, and `df.info()` shows that:

* The dataset contains **1,599 rows** and **12 columns**.
* All columns have **1,599 non-null values**, indicating there are **no missing values**.
* Eleven predictor variables are stored as **`float64`**.
* The target variable, **`quality`**, is stored as **`int64`**, representing the wine quality score assigned by human evaluators.

Since the dataset contains no missing values, no imputation or removal of missing data is required during preprocessing. The next step is to perform exploratory data analysis (EDA) to better understand the distributions of the variables, identify potential outliers, examine relationships between features, and investigate how the physicochemical properties relate to wine quality.